### PR TITLE
Remove unused graphics functions

### PR DESCRIPTION
**Pull request by asarium**
`gf_set_palette` was a no-op for all graphics APIs so I just removed it.

### CODE REVIEW

File: code/graphics/2d.cpp [hunk 1]
```diff
@@ -702,14 +702,10 @@ void gr_set_palette_internal( const char *name, ubyte * palette, int restrict_fo
...
 	if ( Gr_inited ) {
-		if (gr_screen.gf_set_palette) {
-			(*gr_screen.gf_set_palette)(Gr_current_palette, restrict_font_to_128 );
-
-			// Since the palette set code might shuffle the palette,
-			// reload it into the source palette
-			if ( palette ) {
-				memmove( palette, Gr_current_palette, 768 );
-			}
+		// Since the palette set code might shuffle the palette,
```
Comment (MageKing17) [inline code/graphics/2d.cpp]: Is any of this needed if we aren't actually calling any palette-set code?

Comment (asarium) [inline code/graphics/2d.cpp]: I'm not sure but I didn't want to remove it because that may cause issues somewhere else.

**Comment by chief1983** [comment]
So we waiting on this because of other graphics stuff then or what?

**Comment by asarium** [comment]
IMO it's ready to be merged. The code @MageKing17 referenced could probably be removed but AFAIK it's used somewhere and I didn't want to remove that code without being sure if it's needed or not.

**Pull request merged**